src/main/java/de/tum/in/www1/artemis/service/plagiarism/text/TextPlagiarismDetectionService.java

```diff
@@ -0,0 +1,64 @@
+package de.tum.in.www1.artemis.service.plagiarism.text;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.domain.participation.Participation;
+
+@Service
+public class TextPlagiarismDetectionService {
+
+    /**
+     * Convenience method of extract all latest submissions from a TextExercise and compute pair-wise distancse.
+     *
+     * @param exerciseWithParticipationsAndSubmissions Text Exercise with fetched participations and submissions
+     * @param comparisionStrategy TextComparisionStrategy
+     * @return Map of text submission pairs and distance score
+     */
+    public Map<Set<TextSubmission>, Double> compareSubmissionsForExerciseWithStrategy(TextExercise exerciseWithParticipationsAndSubmissions,
+            TextComparisionStrategy comparisionStrategy) {
+        final List<TextSubmission> textSubmissions = textSubmissionsForComparison(exerciseWithParticipationsAndSubmissions);
+        return compareSubmissionsForExerciseWithStrategy(textSubmissions, comparisionStrategy);
+    }
+
+    /**
+     * Pairwise comparision of text submissions using a TextComparisionStrategy
+     *
+     * @param textSubmissions List of text submissions
+     * @param comparisionStrategy TextComparisionStrategy
+     * @return Map of text submission pairs and distance score
+     */
+    public Map<Set<TextSubmission>, Double> compareSubmissionsForExerciseWithStrategy(List<TextSubmission> textSubmissions, TextComparisionStrategy comparisionStrategy) {
+        final Map<Set<TextSubmission>, Double> map = new HashMap<>();
+
+        for (int i = 0; i < textSubmissions.size(); i++) {
```

Comment:
Iterating like that, items are only compared to items behind them in the list.

If there are submissions [a,b,c] and let's say a & c are identical, a would receive a low distance score and c would not be evaluated at all. Is this intended? If yes, I would indicate that somewhere in the documentation.

Reply:
The idea behind the loop is to only compare each pair once.
for `a,b,c` we will compare `a & b`, `a & c`, and `b & c`.
On purpose, this leaves out `a & a`, `b & a`, `b & b`, `c & a`, `c & b`, and `c & c`.